### PR TITLE
Ele 3682 alerts grouping

### PR DESCRIPTION
**Pull request by MikaKerman**
null

### CODE REVIEW

File: elementary/monitor/data_monitoring/alerts/integrations/slack/slack.py [hunk 1]
```diff
@@ -873,6 +878,147 @@ def _get_group_by_table_template(
...
+        bullet_icon: str,
+    ) -> None:
+        if not alerts:
+            return
+
+        section_text_rows = [f"*{sub_title}*"]
+        for alert in alerts:
+            text = f":{bullet_icon}: {alert.summary}"
+            if alert.report_url:
+                text = " - ".join([text, f"<{alert.report_url}|View Details>"])
```
Comment (NoyaArie) [inline elementary/monitor/data_monitoring/alerts/integrations/slack/slack.py:968]: check the url in your alerts, I think the report_url tasks to the dashboard, is this what we want?

**Review by NoyaArie — COMMENTED** [review]
LGTM! Left a few small comments

File: elementary/monitor/data_monitoring/alerts/integrations/base_integration.py [hunk 2]
```diff
@@ -91,12 +101,83 @@ def send_alert(
...
+        flattened_alerts: List[
+            Union[TestAlertModel, ModelAlertModel, SourceFreshnessAlertModel]
+        ] = []
+        for alert in alerts:
+            if isinstance(alert, GroupedAlert):
+                flattened_alerts.extend(alert.alerts)
+            else:
+                flattened_alerts.append(alert)
+
+        if len(flattened_alerts) > threshold:
```
[the remark this answers is not in the document]
Comment (MikaKerman) [reply elementary/monitor/data_monitoring/alerts/integrations/base_integration.py]: Done.

File: elementary/monitor/data_monitoring/alerts/integrations/slack/slack.py (hunk 1 above)
Comment (MikaKerman) [reply elementary/monitor/data_monitoring/alerts/integrations/slack/slack.py]: Done.

File: elementary/monitor/data_monitoring/alerts/integrations/base_integration.py [hunk 3]
```diff
@@ -91,12 +101,83 @@ def send_alert(
...
+        ],
+        group_alerts_threshold: int,
+        *args,
+        **kwargs,
+    ) -> Generator[
+        Tuple[
+            Union[
+                TestAlertModel,
+                ModelAlertModel,
+                SourceFreshnessAlertModel,
```
[the remark this answers is not in the document]
Comment (MikaKerman) [reply elementary/monitor/data_monitoring/alerts/integrations/base_integration.py:165]: If the alert is an AlertsGroup, the inner alerts will be generated. This changes the behavior of GroupedByTableAlerts, but I checked that the method's caller can handle it as well.
https://github.com/elementary-data/elementary/blob/161330daf6c2d0f5619a2898ab5437424b6c3a06/elementary/monitor/data_monitoring/alerts/data_monitoring_alerts.py#L259

**Review by NoyaArie — APPROVED** [review]
LGTM 🎉

**Pull request merged**